src/views/app/detail-panel/conversation-preview-panel.tsx

```diff
@@ -3,51 +3,35 @@
  *
  * SPDX-License-Identifier: AGPL-3.0-only
  */
-import React, { FC, ReactElement, useCallback, useEffect, useMemo } from 'react';
+import React, { ReactElement, useCallback } from 'react';
 
 import { Container, Shimmer } from '@zextras/carbonio-design-system';
-import { FOLDERS, useTags, useUserSettings } from '@zextras/carbonio-shell-ui';
-import { filter, find, map, sortBy } from 'lodash';
-import { useParams } from 'react-router-dom';
+import { map } from 'lodash';
 
 import { ConversationMessagePreview } from './conversation-message-preview';
-import PreviewPanelHeader from './preview/preview-panel-header';
-import { useAppDispatch, useAppSelector } from '../../../hooks/redux';
-import { getConv, searchConv } from '../../../store/actions';
-import {
-	selectConversationExpandedStatus,
-	selectConversationsArray,
-	selectCurrentFolderExpandedStatus
-} from '../../../store/conversations-slice';
-import { selectMessages } from '../../../store/messages-slice';
-import type { Conversation, MailsStateType } from '../../../types';
-import { useExtraWindow } from '../extra-windows/use-extra-window';
+import { useAppSelector } from '../../../hooks/redux';
+import { selectCurrentFolderExpandedStatus } from '../../../store/conversations-slice';
+import type { Conversation } from '../../../types';
 
-export const ConversationPreviewComponent = ({
-	conversation
+export const ConversationPreviewPanel = ({
+	conversation,
+	isInsideExtraWindow,
+	convSortOrder
 }: {
 	conversation: Conversation;
+	isInsideExtraWindow: boolean;
+	convSortOrder: string;
 }): ReactElement => {
-	const settings = useUserSettings();
-	const messages = useAppSelector(selectMessages);
 	const conversationStatus = useAppSelector(selectCurrentFolderExpandedStatus)[conversation.id];
-	const convMessages = useMemo(() => {
-		const msgs = map(conversation?.messages, (item) => messages[item.id] ?? item);
```

Comment:
Is the message equal between the conversation.messages and the messages inside the message slice?
My worry is that the message slice has more data than the conv.messages inside the conversation slice.
Are you sure you can replace this value with the other?

Reply:
conv.messages are used only for the ids. in the ConversationMessagePreview component, where they are passed as props, the full messages are extracted from the message store
https://github.com/zextras/carbonio-mails-ui/pull/523/files#diff-112e1544718bfb26e1666e83f1fd0c775f12aa80fc77ae94bcfa5a55303ad7feR31
in this way we avoid of fetching the whole messages store, which will cause rerenders if messages not of interest to the conversation are update

Reply:
oh ok clear, I missed that part. It could also receive the id only instead of the entire message at this point but i don't think it would change much